### PR TITLE
[Text] Add support for `bodyXs` variant and fix `headingLg` font weight

### PR DESCRIPTION
**Pull request by laurkim**
### WHY are these changes introduced?

Updates references to font tokens in the Text component variants table.
Adds support for new `bodyXs` variant.
Fixes incorrect font weight for `headingLg` variant.

### WHAT is this pull request doing?

Migrates font token values in Text docs from v11 to v12 at `/components/typography/text#variant-tokens`.
Adds support for new `bodyXs` variant.
Fixes incorrect font weight for `headingLg` variant.
    <details>
      <summary>Text variants table — before</summary>
      Text variants table — before
    </details>
    <details>
      <summary>Text variants table — after</summary>
      Text variants table — after
    </details>

### How to 🎩

🌀 [Spin](https://admin.web.text-headingxl-font-change.lo-kim.us.spin.dev/store/shop1) 
[Storybook](https://5d559397bae39100201eedc1-wekthcljix.chromatic.com/?path=/story/all-components-text--variants)
[Prod storybook](https://storybook.polaris.shopify.com/?path=/story/all-components-text--variants)

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris.shopify.com/content/components/typography/text.md [hunk 1]
```diff
@@ -68,15 +68,15 @@ Each variant uses a predetermined combination of the [font tokens](/tokens/font)
...
-| `heading2xl` | `--p-font-size-500` | 28       | 1.75      | `--p-line-height-5`    | `--p-font-weight-semibold` | Yes       |
-| `headingXl`  | `--p-font-size-400` | 24       | 1.5       | `--p-line-height-4`    | `--p-font-weight-semibold` | Yes       |
-| `headingLg`  | `--p-font-size-300` | 20       | 1.25      | `--p-line-height-3`    | `--p-font-weight-semibold` | Yes       |
-| `headingMd`  | `--p-font-size-200` | 16       | 1         | `--p-line-height-3`    | `--p-font-weight-semibold` | No        |
-| `headingSm`  | `--p-font-size-100` | 14       | 0.875     | `--p-line-height-2`    | `--p-font-weight-semibold` | No        |
-| `bodyLg`     | `--p-font-size-200` | 16       | 1         | `--p-line-height-2`    | `--p-font-weight-regular`  | No        |
-| `bodyMd`     | `--p-font-size-100` | 14       | 0.875     | `--p-line-height-2`    | `--p-font-weight-regular`  | No        |
-| `bodySm`     | `--p-font-size-75`  | 12       | 0.75      | `--p-line-height-1`    | `--p-font-weight-regular`  | No        |
+| `heading3xl` | `--p-font-size-900` | 36       | 2.25      | `--p-line-height-1000` | `--p-font-weight-bold`     | Yes       |
```
Comment (sarahill) [inline polaris.shopify.com/content/components/typography/text.md]:
```suggestion
| `heading3xl` | `--p-font-size-900` | 36       | 2.25      | `--p-line-height-1200` | `--p-font-weight-bold`     | Yes       |
```

updated line-height

File: polaris.shopify.com/content/components/typography/text.md [hunk 2]
```diff
@@ -68,15 +68,15 @@ Each variant uses a predetermined combination of the [font tokens](/tokens/font)
...
+| `heading2xl` | `--p-font-size-750` | 30       | 1.875     | `--p-line-height-800`  | `--p-font-weight-bold`     | Yes       |
```
Comment (sarahill) [inline polaris.shopify.com/content/components/typography/text.md]:
```suggestion
| `heading2xl` | `--p-font-size-750` | 30       | 1.875     | `--p-line-height-1000`  | `--p-font-weight-bold`     | Yes       |
```

updated line-height

File: polaris.shopify.com/content/components/typography/text.md [hunk 3]
```diff
@@ -68,15 +68,15 @@ Each variant uses a predetermined combination of the [font tokens](/tokens/font)
...
+| `headingXl`  | `--p-font-size-600` | 24       | 1.5       | `--p-line-height-700`  | `--p-font-weight-bold`     | Yes       |
```
Comment (sarahill) [inline polaris.shopify.com/content/components/typography/text.md]:
```suggestion
| `headingXl`  | `--p-font-size-600` | 24       | 1.5       | `--p-line-height-800`  | `--p-font-weight-bold`     | Yes       |
```

updated line-height

File: polaris.shopify.com/content/components/typography/text.md [hunk 4]
```diff
@@ -68,15 +68,15 @@ Each variant uses a predetermined combination of the [font tokens](/tokens/font)
...
+| `headingLg`  | `--p-font-size-500` | 20       | 1.25      | `--p-line-height-600`  | `--p-font-weight-bold`     | Yes       |
```
Comment (sarahill) [inline polaris.shopify.com/content/components/typography/text.md]:
```suggestion
| `headingLg`  | `--p-font-size-500` | 20       | 1.25      | `--p-line-height-600`  | `--p-font-weight-semibold`     | Yes       |
```

updated font-weight

File: polaris.shopify.com/content/components/typography/text.md [hunk 5]
```diff
@@ -68,15 +68,15 @@ Each variant uses a predetermined combination of the [font tokens](/tokens/font)
...
+| `headingMd`  | `--p-font-size-400` | 16       | 1         | `--p-line-height-600`  | `--p-font-weight-semibold` | No        |
```
Comment (sarahill) [inline polaris.shopify.com/content/components/typography/text.md]:
```suggestion
| `headingMd`  | `--p-font-size-350` | 14       | .875         | `--p-line-height-500`  | `--p-font-weight-semibold` | No        |
```

updated size and line-height

File: polaris.shopify.com/content/components/typography/text.md [hunk 6]
```diff
@@ -68,15 +68,15 @@ Each variant uses a predetermined combination of the [font tokens](/tokens/font)
...
+| `headingSm`  | `--p-font-size-350` | 14       | 0.875     | `--p-line-height-500`  | `--p-font-weight-semibold` | No        |
```
Comment (sarahill) [inline polaris.shopify.com/content/components/typography/text.md]:
```suggestion
| `headingSm`  | `--p-font-size-325` | 13       | 0.8125     | `--p-line-height-500`  | `--p-font-weight-semibold` | No        |
```

File: polaris.shopify.com/content/components/typography/text.md [hunk 7]
```diff
@@ -68,15 +68,15 @@ Each variant uses a predetermined combination of the [font tokens](/tokens/font)
...
+| `bodyLg`     | `--p-font-size-400` | 16       | 1         | `--p-line-height-500`  | `--p-font-weight-regular`  | No        |
```
Comment (sarahill) [inline polaris.shopify.com/content/components/typography/text.md]:
```suggestion
| `bodyLg`     | `--p-font-size-350` | 14       | .875         | `--p-line-height-500`  | `--p-font-weight-regular`  | No        |
```

File: polaris.shopify.com/content/components/typography/text.md [hunk 8]
```diff
@@ -68,15 +68,15 @@ Each variant uses a predetermined combination of the [font tokens](/tokens/font)
...
+| `bodyMd`     | `--p-font-size-350` | 14       | 0.875     | `--p-line-height-500`  | `--p-font-weight-regular`  | No        |
```
Comment (sarahill) [inline polaris.shopify.com/content/components/typography/text.md]:
```suggestion
| `bodyMd`     | `--p-font-size-325` | 13       | 0.8125     | `--p-line-height-500`  | `--p-font-weight-regular`  | No        |
```

File: polaris.shopify.com/content/components/typography/text.md [hunk 9]
```diff
@@ -68,15 +68,15 @@ Each variant uses a predetermined combination of the [font tokens](/tokens/font)
...
+| `bodySm`     | `--p-font-size-300` | 12       | 0.75      | `--p-line-height-400`  | `--p-font-weight-regular`  | No        |
```
Comment (sarahill) [inline polaris.shopify.com/content/components/typography/text.md:79]:
```suggestion
| `bodySm`     | `--p-font-size-300` | 12       | 0.75      | `--p-line-height-400`  | `--p-font-weight-regular`  | No        |
| `bodyXs`     | `--p-font-size-275` | 11       | 0.6875      | `--p-line-height-300`  | `--p-font-weight-regular`  | No        |
```

**Review by sarahill — DISMISSED** [review]

**Comment by sarahill** [comment]
Made a couple suggested updates based on Figma and [this file](https://github.com/Shopify/polaris/blob/main/polaris-tokens/src/themes/base/text.ts) as reference :)

**Comment by laurkim** [comment]
> Made a couple suggested updates based on Figma and [this file](https://github.com/Shopify/polaris/blob/main/polaris-tokens/src/themes/base/text.ts) as reference :)

Oops, I accidentally referenced the values for the xs-sm breakpoints 😅 
Thanks for catching all of these @sarahill! ❤️

File: polaris.shopify.com/content/components/typography/text.md (hunk 4 above)
Comment (laurkim) [reply polaris.shopify.com/content/components/typography/text.md]: @sarahill It looks like in the [Text styles](https://github.com/Shopify/polaris/blob/ed5d7d6bfb881d2b74a3c701fa6e3506cffd60b2/polaris-react/src/components/Text/Text.scss#L78) `headingLg` is using `--p-font-weight-bold`. I went through the commit history and looks like that's what we set it to during uplift. I just want to confirm that `--p-font-weight-semibold` is what we want for the weight for this variant, I can fix it in this PR if so! 👍

File: polaris.shopify.com/content/components/typography/text.md (hunk 9 above)
Comment (laurkim) [reply polaris.shopify.com/content/components/typography/text.md:79]: @sarahill we don't support `bodyXs` as a variant in Text right now. 
Is that something that should be added in?

Comment (sarahill) [reply polaris.shopify.com/content/components/typography/text.md:79]: Ah yea that should be added to the `Text` component. We must have just missed it.

File: polaris.shopify.com/content/components/typography/text.md (hunk 4 above)
Comment (sarahill) [reply polaris.shopify.com/content/components/typography/text.md]: Yea `headingLg` should be `semibold`

Comment (laurkim) [reply polaris.shopify.com/content/components/typography/text.md]: Gotcha, will update in this PR and change the context of these changes a bit 👍

File: polaris.shopify.com/content/components/typography/text.md (hunk 9 above)
Comment (laurkim) [reply polaris.shopify.com/content/components/typography/text.md:79]: Gotcha, will add in this PR 👍

**Review by sarahill — APPROVED** [review]
🎉

**Review by kyledurand — APPROVED** [review]

**Pull request merged**